[master] Fixed issue with add_store and del_store

HarshalSawant1234 opened the pull request:
### Enable saltenv in add_store and del_store

### What issues does this PR fix or reference?
Fixes:
Certificate not found error when running state with saltenv other than base

### Previous Behavior
local:
----------
          ID: security_deploy_certificates
    Function: certutil.add_store
        Name: salt://base/files/certificates/wildcard.cer
      Result: False
     Comment: Certificate file not found.
     Started: 11:09:30.709032
    Duration: 31.246 ms
     Changes:

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:  31.246 ms

### New Behavior
local:
----------
          ID: security_deploy_itsb_certificates
    Function: certutil.add_store
        Name: salt://base/files/certificates/wildcard.cer
      Result: True
     Comment:
     Started: 11:47:55.578772
    Duration: 263.524 ms
     Changes:
              ----------
              added:
                  salt://base/files/certificates/wildcard.cer

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time: 263.524 ms

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

HarshalSawant1234 committed:
Add missing saltenv tag which was causing file not found error in the module

State was only looking into the base saltenv for certificate file when add_store or del_store was called.

HarshalSawant1234 committed:
Fixed module not loading when called from its respective state

FIX: SyntaxError positional argument follows keyword argument

Ch3LL commented on salt/modules/win_certutil.py:
```diff
@@ -122,7 +122,7 @@ def add_store(source, store, retcode=False, saltenv="base"):
 
 
-def del_store(source, store, retcode=False, saltenv="base"):
+def del_store(source, store, saltenv="base", retcode=False):
```
Instead of changing the position of the kwargs, can we just call the kwarg from the call in the state. I'll include example in the state module below.

twangboy replied:
This ^^^

Ch3LL commented on salt/states/win_certutil.py:
```diff
@@ -65,7 +65,7 @@ def add_store(name, store, saltenv="base"):
         return ret
 
-    cert_serial = __salt__["certutil.get_cert_serial"](name)
+    cert_serial = __salt__["certutil.get_cert_serial"](name, saltenv)
```
```suggestion
    cert_serial = __salt__["certutil.get_cert_serial"](name, saltenv=saltenv)
```

This would also apply to the other calls below

Ch3LL reviewed (changes requested):
This will also require a changelog and test coverage

twangboy reviewed (changes requested):
Please create a changelog and address comments form @Ch3LL

dwoz committed:
Address review feedback: use saltenv kwarg, fix del_store, add tests

- Revert the parameter-order swap in win_certutil module (keep
  retcode before saltenv to avoid breaking callers using positional
  arguments) per twangboy/Ch3LL review feedback
- Pass saltenv=saltenv as a keyword argument in all four state-to-module
  calls so non-base saltenvs are honoured during cert resolution
- Fix the previously-missed get_cert_serial call in del_store (it was
  still hardcoded to saltenv="base")
- Add regression tests verifying saltenv propagation in both
  add_store and del_store state functions
- Add changelog fragment (64659.fixed.md)

twangboy commented:
Please fix this on the earliest branch where the bug exists... probably 3006.x
